clients/ctl/admin-ui/src/features/config-wizard/PrivacyDeclarationForm.tsx

```diff
@@ -55,20 +49,35 @@ import {
   useGetSystemByFidesKeyQuery,
   useUpdateSystemMutation,
 } from "../system/system.slice";
-import { changeReviewStep, selectSystemFidesKey } from "./config-wizard.slice";
+import PrivacyDeclarationAccordion from "./PrivacyDeclarationAccordion";
 
-type FormValues = Partial<PrivacyDeclaration>;
+type FormValues = PrivacyDeclaration;
 
-const PrivacyDeclarationForm = ({
-  handleCancelSetup,
-}: {
-  handleCancelSetup: () => void;
-}) => {
-  const systemFidesKey = useAppSelector(selectSystemFidesKey);
-  const { data: existingSystem } = useGetSystemByFidesKeyQuery(systemFidesKey);
+const ValidationSchema = Yup.object().shape({
+  name: Yup.string().required().label("Declaration name"),
+  data_categories: Yup.array(Yup.string())
+    .min(1, "Must assign at least one data category")
+    .label("Data categories"),
+  data_use: Yup.string().required().label("Data use"),
+  data_subjects: Yup.array(Yup.string())
+    .min(1, "Must assign at least one data subject")
+    .label("Data subjects"),
+});
+
+interface Props {
+  systemKey: System["fides_key"];
+  onCancel: () => void;
+  onSuccess: (system: System) => void;
+}
+
+const PrivacyDeclarationForm = ({ systemKey, onCancel, onSuccess }: Props) => {
+  const { data: existingSystem, isLoading: isLoadingSystem } =
+    useGetSystemByFidesKeyQuery(systemKey);
```

Comment:
With these components having most of their state dependence extracted, I would be a bit tempted to remove these queries as well: instead of taking the key and querying the system, take the system itself and the parent does the query. Not a big deal, though, seeing as that data is cached.

Reply:
good idea!